sshconnect2.c

```diff
@@ -1624,6 +1636,36 @@ key_type_allowed_by_config(struct sshkey *key)
 	return 0;
 }
 
+/* obtain a list of keys from the agent */
+static int
+get_agent_identities(struct ssh *ssh, int *agent_fdp,
+    struct ssh_identitylist **idlistp)
+{
+	int r, agent_fd;
+	struct ssh_identitylist *idlist;
+
+	if ((r = ssh_get_authentication_socket(&agent_fd)) != 0) {
+		if (r != SSH_ERR_AGENT_NOT_PRESENT)
+			debug_fr(r, "ssh_get_authentication_socket");
+		return r;
+	}
+	if ((r = ssh_agent_bind_hostkey(agent_fd, ssh->kex->initial_hostkey,
```

Comment:
what will happen if the user runs an old agent that dislikes the bind-message? should be be able to disable binding?

Reply:
openssh ssh-agent would just return SSH_AGENT_FAILURE for unknown messages and we'd continue without error.

It's possible that other agent implementations could behave badly, but the GnomeKeyring ssh-agent implementation seems to be okay (it forwards unknown requests to an existing `$SSH_AUTH_SOCK`, presumably a real ssh-agent). gnupg's gpg-agent seems to [behave like OpenSSH](https://github.com/gpg/gnupg/blob/965bb0693c0df04e6dbd57a8b471944367cf6c4e/agent/command-ssh.c#L3473)

Reply:
Likewise, it seems to be a soft error for [PuTTY](https://github.com/github/putty/blob/master/pageant.c#L1291) too

Reply:
(So I'm inclined to **not** add another knob for this ahead of time)